driver-examples/src/main/java/com/datastax/driver/examples/json/codecs/JacksonJsonCodec.java

```diff
@@ -0,0 +1,155 @@
+import com.datastax.driver.examples.json.exceptions.InvalidTypeException;
+import com.datastax.oss.driver.api.core.ProtocolVersion;
+import com.datastax.oss.driver.api.core.type.DataType;
+import com.datastax.oss.driver.api.core.type.DataTypes;
+import com.datastax.oss.driver.api.core.type.codec.TypeCodec;
+import com.datastax.oss.driver.api.core.type.reflect.GenericType;
+import com.datastax.oss.driver.internal.core.util.Strings;
+import com.datastax.oss.protocol.internal.util.Bytes;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * A JSON codec that uses the <a href="http://wiki.fasterxml.com/JacksonHome">Jackson</a> library to
+ * perform serialization and deserialization of JSON objects.
+ *
+ * <p>This codec maps a single Java object to a single JSON structure at a time; mapping of arrays
+ * or collections to root-level JSON arrays is not supported, but such a codec can be easily crafted
+ * after this one.
+ *
+ * <p>Note that this codec requires the presence of Jackson library at runtime. If you use Maven,
+ * this can be done by declaring the following dependency in your project:
+ *
+ * <p>
+ *
+ * <pre>{@code
+ * <dependency>
+ *   <groupId>com.fasterxml.jackson.core</groupId>
+ *   <artifactId>jackson-databind</artifactId>
+ *   <version>2.6.3</version>
+ * </dependency>
+ * }</pre>
+ */
+public class JacksonJsonCodec<T> implements TypeCodec<T> {
+
+  private final ObjectMapper objectMapper;
+  private final GenericType<T> javaType;
+
+  /**
+   * Creates a new instance for the provided {@code javaClass}, using a default, newly-allocated
+   * {@link ObjectMapper}.
+   *
+   * @param javaClass the Java class this codec maps to.
+   */
+  public JacksonJsonCodec(Class<T> javaClass) {
+    this(javaClass, new ObjectMapper());
+  }
+
+  /**
+   * Creates a new instance for the provided {@code javaClass}, and using the provided {@link
+   * ObjectMapper}.
+   *
+   * @param javaClass the Java class this codec maps to.
+   */
+  public JacksonJsonCodec(Class<T> javaClass, ObjectMapper objectMapper) {
+    this.javaType = GenericType.of(javaClass);
+    this.objectMapper = objectMapper;
+  }
+
+  @NonNull
+  @Override
+  public GenericType<T> getJavaType() {
+    return javaType;
+  }
+
+  @NonNull
+  @Override
+  public DataType getCqlType() {
+    return DataTypes.TEXT;
+  }
+
+  @Nullable
+  @Override
+  public ByteBuffer encode(@Nullable T value, @NonNull ProtocolVersion protocolVersion) {
+    if (value == null) return null;
+    try {
+      return ByteBuffer.wrap(objectMapper.writeValueAsBytes(value));
+    } catch (JsonProcessingException e) {
+      throw new InvalidTypeException(e.getMessage(), e);
```

Comment:
In driver 4.0 we don't have any specific exception for codecs except `CodecNotFoundException`. When a codec cannot serialize or deserialize something, we use `IllegalArgumentException`. I don't think you need to introduce a special exception in these examples.